### PR TITLE
D8CORE-2709 Entity Reference Autocomplete & Select support

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Provided react widget for entity reference autocomplete and select list widgets

# Need Review By (Date)
- 10/14

# Urgency
- medium

# Steps to Test
1. Checkout this branch and clear browser caches
1. create/edit a paragraph type and add an entity reference field to reference other content. You can use any cardinality.
1. In the paragraph type form display settings, select either the autocomplete or select list widgets
1. create a basic page and validate the field you created works and saves the data correctly.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
I'm going to look at this today.

File: js/src/Components/Widgets/AutocompleteReferenceWidget.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {FormHelperText} from "@material-ui/core";
+import TextField from '@material-ui/core/TextField';
+import FormGroup from '@material-ui/core/FormGroup';
+import Autocomplete from '@material-ui/lab/Autocomplete';
```
Comment (sherakama) [inline js/src/Components/Widgets/AutocompleteReferenceWidget.js:5]: Handy

**Comment by sherakama** [comment]
Overall this is working well. Even works with the media library widget 👍 

A couple of things:
1. Default values are not being populated
2. Better check in with @ishahism on the field UI. I've not seen the select option with the `x` before on the single select
3. Entity Reference Revisions don't seem to be supported but that is to be expected. 

Screen Shot 2020-10-07 at 11 58 04 AM
Screen Shot 2020-10-07 at 11 58 23 AM
Screen Shot 2020-10-07 at 12 01 27 PM
Screen Shot 2020-10-07 at 12 02 00 PM

**Review by sherakama — APPROVED** [review]
Code works well. Just a quick check in with UI/UX.

**Comment by pookmish** [comment]
I actually did a meeting with Ishi and she's good with this approach for now. She would like a more intuitive way to change the order of the items, but we don't know if we'll have more than 1 cardinality at this time.

The default values are the defaults as defined in the field settings right? I didn't catch those. Most of our widgets currently don't do that either. Should we create a separate ticket to work on each of those?

File: js/src/Components/Widgets/AutocompleteReferenceWidget.js (hunk 1 above)
Comment (sherakama) [reply js/src/Components/Widgets/AutocompleteReferenceWidget.js:5]: > The default values are the defaults as defined in the field settings right? I didn't catch those. Most of our widgets currently don't do that either. Should we create a separate ticket to work on each of those?

Yeah, I think a separate issue is the right way to go. I don't think it is working anywhere yet.

**Comment by sherakama** [comment]
> The default values are the defaults as defined in the field settings right? I didn't catch those. Most of our widgets currently don't do that either. Should we create a separate ticket to work on each of those?

A new ticket is the right way to go. I don't think it is working anywhere yet.

**Comment by pookmish** [comment]
https://stanfordits.atlassian.net/browse/HSD8-922 for the defaults values

**Pull request merged**